paper-api/src/main/java/org/bukkit/entity/Player.java

```diff
@@ -3275,6 +3275,22 @@ default boolean hasResourcePack() {
      */
     java.util.Locale locale();
     // Paper end
+
+    /**
+     * Returns the locale the server will use to send messages translated via the Adventure global translator.
+     * By default, the value of {@link Player#locale()} is used.
+     *
+     * @return the locale.
+     */
+    java.util.Locale getEffectiveLocale();
+
+    /**
+     * Change the locale the server will be translating its messages to.
```

Comment:
I'd maybe also mention that the effective locale will override any changes from the client as well.
E.g. if a plugin reads a players locale as en_us, sets their effective locale to en_gb, the player changing their locale to de_de will not have an affect on the effective local.